hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java

```diff
@@ -597,4 +603,12 @@ void handleNodeSlowness(RaftGroup group, RoleInfoProto roleInfoProto) {
   void handleNoLeader(RaftGroup group, RoleInfoProto roleInfoProto) {
     handlePipelineFailure(group.getGroupId(), roleInfoProto);
   }
+
+  void handleInstallSnapshotFromLeader(RaftGroup group,
+      RoleInfoProto roleInfoProto, TermIndex firstTermIndexInLog) {
+    LOG.warn("Install snapshot notification received from Leader with " +
+        "termIndex : " + firstTermIndexInLog +
+        ", terminating pipeline " + group.getGroupId());
+    handlePipelineFailure(group.getGroupId(), roleInfoProto);
```

Comment:
I am trying to understand this change, Jira says disable install snapshot for ContainerStateMachine.
I see that setSnapShotEnabled is set to false.
But what is the reason for handleInstallSnapShotFromLeader to call handlePipelineFailure.

Sorry, I don't have complete context of this, can you provide some explanation of this change, and why it is done, from which I can understand better.

Reply:
Yes sorry, let me add to the description.